### DISCUSSION TITLE
Can it add collapsible effect feature?

### DISCUSSION THREAD

**Discussion by sanshigongA** [eclipse-birt/birt#1756] [Q&A]
![动画](https://github.com/eclipse-birt/birt/assets/107231757/38e76c9f-353d-48ba-a68c-dc3b0e60f4aa)

**Comment by speckyspooky** [comment]
On excel level you would get this kind of grouping with grouped rows.

If you want it on the viewer level, yes it is possible with native JavaScript-code. I created such small JavaScript for some of my report previews.
The steps draft would have the following steps:
- grouping: necessary to have the group label with the click-icon
- bookmark: each row must be get an ID based on the boomark
- row id: the ID must be have a kind of pattern for later JavaScript fetch like this
group 01: "group-01-row-01","group-01-row-02","group-01-row-...", "group-01-row-10" 
group 02: "group-02-row-01","group-02-row-02","group-02-row-...", "group-02-row-10"
- click icon: is an Text-element of HTML with script-tag to call the function "handleRowView('group-02')
- handleRowView - function draft:


```
function handleRowView( groupKey ) {
    var nodeRowList = document.querySelectorAll("[id^='" + groupKey + "']");
    for (nodeRow of nodeRowList) {
        if (typeof nodeRow.style.display == "undefined" || nodeRow.style.display == "") {
             nodeRow.style.display = "none";  // hide the rows
        } else {
             nodeRow.style.display = "";  // show the rows
        }
    }
}
```
- of course you need some additional lines to change your icon to the "minus"- / "plus"-sign
